core/src/impl/Kokkos_ViewCtor.hpp

```diff
@@ -325,29 +325,9 @@ KOKKOS_FUNCTION const auto &get_property(
 
 template <typename Tag, typename... P>
 KOKKOS_FUNCTION auto &get_property(ViewCtorProp<P...> &view_ctor_prop) {
-  if constexpr (std::is_same_v<Tag, ExecutionSpaceTag>) {
-    static_assert(ViewCtorProp<P...>::has_execution_space);
-    using execution_space_type = typename ViewCtorProp<P...>::execution_space;
-    return static_cast<ViewCtorProp<void, execution_space_type> &>(
-               view_ctor_prop)
-        .value;
-  } else if constexpr (std::is_same_v<Tag, MemorySpaceTag>) {
-    static_assert(ViewCtorProp<P...>::has_memory_space);
-    using memory_space_type = typename ViewCtorProp<P...>::memory_space;
-    return static_cast<ViewCtorProp<void, memory_space_type> &>(view_ctor_prop)
-        .value;
-  } else if constexpr (std::is_same_v<Tag, LabelTag>) {
-    static_assert(ViewCtorProp<P...>::has_label);
-    return static_cast<ViewCtorProp<void, std::string> &>(view_ctor_prop).value;
-  } else if constexpr (std::is_same_v<Tag, PointerTag>) {
-    static_assert(ViewCtorProp<P...>::has_pointer);
-    using pointer_type = typename ViewCtorProp<P...>::pointer_type;
-    return static_cast<ViewCtorProp<void, pointer_type> &>(view_ctor_prop)
-        .value;
-  } else {
-    static_assert(std::is_same_v<Tag, void>, "Invalid property tag!");
-    return view_ctor_prop;
-  }
+  const ViewCtorProp<P...> &tmp = view_ctor_prop;
+  const auto &tmp_return        = get_property<Tag>(tmp);
+  return const_cast<std::decay_t<decltype(tmp_return)> &>(tmp_return);
```

Comment:
Is that a joke?

Reply:
I don't think we should have a non-const reference version of get_property in the first place. I don't think there is any reason for this ting to not be immutable?

Reply:
We actually need both versions

Reply:
[masterleinad/kokkos@11842ae/core/src/Kokkos_CopyViews.hpp#L3820-L3821](https://github.com/masterleinad/kokkos/blob/11842ae0f15b6fe2563b18c106f5177cfcd86570/core/src/Kokkos_CopyViews.hpp#L3820-L3821) is an example where we modify a property by first extracting it from a ViewCtorProp<> argument using get_propetry.

Reply:
More constructively, would it make sense to define the const version in terms of the non-const version?